veps/sig-compute/pcihole64-proposal.md

```diff
@@ -0,0 +1,296 @@
+# VEP #33: Allow to disable the 64-bit PCI hole
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone /
+release*.
+
+- [x] (R) Enhancement issue created, which links to VEP dir
+  in [kubevirt/enhancements] (not the initial VEP PR)
+
+## Overview
+
+This proposal focuses on allowing to disable the 64-bit PCI hole in
+`VirtualMachines` (VM), so that legacy operating systems (OS) like Windows
+XP or Server 2k3 can run on KubeVirt.
+
+An explanation of the 64-bit PCI hole can be found
+[here](https://en.wikipedia.org/wiki/PCI_hole).
+
+## Motivation
+
+At the moment legacy OSes like Windows XP or Server 2k3 do not work out of the
+box on KubeVirt. This prevents big organizations running many VMs of all kinds
+from adopting KubeVirt. To allow greater adoption of KubeVirt, this proposal
+aims to make these legacy OSes work on KubeVirt.
+
+For more information on the topic
+see [this thread on kubevirt-dev](https://groups.google.com/g/kubevirt-dev/c/1Ue4brPoc1g)
+and [this issue](https://bugzilla.redhat.com/show_bug.cgi?id=990418)
+
+## Goals
+
+- A way of expressing that the 64-Bit PCI hole should be disabled is added to
+  the KubeVirt API in the most OS-independent way possible.
+- Legacy OSes like Windows XP or Server 2k3 can run on KubeVirt in a
+  supportable way.
+
+## Non Goals
+
+- Committing to a list of supported OSes in KubeVirt.
+- Adding OS specific configurables to the KubeVirt API.
+
+## Definition of Users
+
+- VM owners
+- Namespace owners
+- Cluster admins
+
+## User Stories
+
+- As a VM owner, I want to run legacy OSes like Windows XP or Server 2k3 on
+  KubeVirt.
+- As a VM owner, I want to migrate existing VMs with legacy OSes to KubeVirt
+  and to continue running them with the least possible amount of required
+  modifications.
+- As a namespace owner or cluster admin, I want to provide
+  `VirtualMachinePreferences` or `VirtualMachineClusterPreferences` for legacy
+  OSes like Windows XP or Server 2k3.
+
+## Repos
+
+- [https://github.com/kubevirt/kubevirt](https://github.com/kubevirt/kubevirt)
+
+## Design
+
+Users that want to run legacy OSes do not necessarily need to know what `64-Bit
+PCI hole` means. Instead, they should be able to express that they want to run
+a legacy OSes without worrying about the details. Annotations provide a simple
+way of enabling configuration tweaks or compatibility behavior while avoiding
+modelling specific fields in the core API and becoming too technical for users.
+
+This proposal suggests to add the `kubevirt.io/legacyWindowsCompat` annotation
+to `VirtualMachine` objects, which will trigger KubeVirt to make the required
+configuration changes to allow running legacy OSes. By using an annotation the
+impact to the core API is kept to a minimum and there is room for future changes
+to this feature independently of the core API.
```

Comment:
`For example the implementation could be moved from kubevirt into a  trusted or certified plugin.`